packages/squiggle-lang/src/rescript/Utility/E/E_A.res

```diff
@@ -323,32 +323,45 @@ module Floats = {
       let continuous: array<float> = []
       let discrete = FloatFloatMap.empty()
 
-      let addData = (count: int, value: float): unit => {
-        if count >= minDiscreteWeight {
-          FloatFloatMap.add(value, count->Belt.Int.toFloat, discrete)
+      // In a run of exactly minDiscreteWeight, the first and last
+      // element indices differ by minDistance.
+      let minDistance = minDiscreteWeight - 1
+
+      let len = length(sortedArray)
+      let i = ref(0)
+      while i.contents < len - minDistance {
+        let value = sortedArray[i.contents]
+        if value != sortedArray[i.contents + minDistance] {
+          Js.Array2.push(continuous, value)->ignore
+          i := i.contents + 1
```

Comment:
Tiny optimization, but I assume at this point you could push all of the same values up to sortedArray[i.contents + minDistance], to `continuous`.

Reply:
Thinking about it more, this is probably not worth doing, it would add too much complexity.

Reply:
You can't, since a run might start anywhere in the middle: if you have 5 6 6 6 6 6 and test with `minDistance` of 3, you see that 5 doesn't start a run but the next 6 actually does. It's possible to skip values if you use a smaller stride. I'll explain this as a top-level comment.

Reply:
I said same value. 

If you have 55588888, and line 334 catches, then you rule all of the next 5s out. So start a second loop and increment until you get to a value that's not a 5.

Reply:
One annoying thing is that I presume we need to accept some heuristic of what the data would look like. If values were very likely on being unique, conditional on not having minDiscreteWeight duplicates, then the current code is probably close to optimal.

Reply:
Oh, got it. It's better to search for values not equal to the last one, rather than those that are equal to the first one. The test in the second loop would run at pretty much the same speed as the main one, and there's a hard-to-predict branch when it stops, so I don't think it would be an improvement with a forwards loop. If you run the loop backwards it's fairly similar to my `minDiscreteWeight / 2` version, but splitting it up as `minDiscreteWeight - 1` and `1` instead.